### PR TITLE
Documentation cleanup + Setup with OpenJDK12

### PR DESCRIPTION
**Pull request by m0ru**
Unified all documentation dealing with the setup into one, non-redundant list. Removed deprecated documentation.

Did this while setting up on my system with openjdk12, which required me to make some `javax`-dependencies explicit in the `pom.xml`s. If necessary, i can split those changes off into a separate pull-request.

### CODE REVIEW

File: documentation/configuration-logging.md [hunk 1]
```diff
@@ -4,24 +4,27 @@ Logging in our project is done with **Simple Logging Facade for Java (SLF4J)** a
...
 The following logging levels are used:
-* **ERROR**: if the application fails to start up, or doesnt't work at all;
-* **WARN**: situations where methods can't execute as they are to be expected (methods called in states where they are not allowed, parametes pointing to not existing atoms, etc.) but the application is still in a operative state; calls, data structures, etc., which go not conform with guidelines, good practices or hard constraints.
-* **INFO**: only in infrequent calls; one time initializations; atom status changes (protocol calls);
-* **DEBUG**: everything else.
+
+- **ERROR**: if the application fails to start up, or doesnt't work at all;
+- **WARN**: situations where methods can't execute as they are to be expected (methods called in states where they are not allowed, parametes pointing to not existing atoms, etc.) but the application is still in a operative state; calls, data structures, etc., which go not conform with guidelines, good practices or hard constraints.
+- **INFO**: only in infrequent calls; one time initializations; atom status changes (protocol calls);
+- **DEBUG**: everything else.
```
Comment (quasarchimaere) [inline documentation/configuration-logging.md]: **TRACE** is also available, descriptions seem rather odd(especially error) idk if we want to have any of this in our documentation even, these describe very common loglevels only and depend on the specific logger method call chosen by the developer and not the severity of a problem. i think a reference to log4j or loglevel documentation is more than sufficient

File: documentation/protocol/message-references.md [hunk 2]
```diff
@@ -1,74 +1,80 @@
...
 <messageUri> msg:previousMessage <previousMessageUri>
 ```
-Where `messageUri` is the URI of the message currently being processed and 
-```previousMessageUri``` is the URI of the referenced message. 
+
+Where `messageUri` is the URI of the message currently being processed and
+`previousMessageUri` is the URI of the referenced message.
 In addition to this triple, a signature reference is added to the message:
 
-*TODO: add signature reference*
```
Comment (sigpie) [inline documentation/protocol/message-references.md]: visible TODO comment

File: documentation/protocol/message-references.md [hunk 3]
```diff
@@ -1,74 +1,80 @@
...
-happening before `B`, while they may, on `Pa`'s counterpart `Pb`'s WoN node 
-as `B` happening before `A`. This may happen independently of which participants 
+
+_Note:_ due to network latency and asynchronous processing, it is possible that two
+messages `A` and `B` may be represented on particpant `Pa`'s WoN node as `A`
+happening before `B`, while they may, on `Pa`'s counterpart `Pb`'s WoN node
+as `B` happening before `A`. This may happen independently of which participants
 send those messages. In order to detect such a case and take measures to avoid
 confusion, both participants should be able to access each other's messages.
-*TODO* there may be a way to detect such a case solely based on message references.
```
Comment (sigpie) [inline documentation/protocol/message-references.md]: Visible TODO

File: documentation/protocol/message-references.md [hunk 4]
```diff
@@ -1,74 +1,80 @@
...
+   at least one path to the atom's `Create` message
 1. A Response must always reference the message that it is a response to.
 
-*TODO explain AtomCreatedNotificationMessage*
+_TODO explain AtomCreatedNotificationMessage_
+
+## Invariants in SPARQL
 
-##  Invariants in SPARQL
 SPARQL queries to test message structure invariants
```
Comment (sigpie) [inline documentation/protocol/message-references.md]: This needs more whitespace to be legible.

File: webofneeds/won-docker/README.md [hunk 5]
```diff
@@ -13,45 +14,39 @@ the webofneeds applications as docker containers. There are images for the follo
 - gencert: a tool that can generate certificates for wonnode and owner application
 - bots: bots are used to test the communication between the servers in the application
 
-For deployment of these components we use [docker-compose](https://docs.docker.com/compose/). It is possible to deploy
-the components on multiple servers. For straight forward deployment however we have provided one docker-compose file
-which can be used to deploy all components on one local server.
+### Steps
+
+1. Download and install docker (https://www.docker.com)
```
Comment (sigpie) [inline webofneeds/won-docker/README.md]: Merge steps 1 and 2

File: webofneeds/won-docker/README.md [hunk 6]
```diff
@@ -13,45 +14,39 @@ the webofneeds applications as docker containers. There are images for the follo
...
+   - PowerShell: [deploy_powershell.ps1](/webofneeds/won-docker/deploy/local_image/deploy_powershell.ps1)
+4. After Docker downloading and starting the components you can access the owner and the node here:
+   - owner: [https://10.0.75.1:8082/owner](https://10.0.75.1:8082/owner)
+   - wonnode: [https://10.0.75.1:8889/won](https://10.0.75.1:8889/won)
 
-### How to deploy and run all webofneeds components on a single server
+### Troubleshooting based on steps
 
-**We recommend using the quick guide [tutorial](/documentation/how-to-won/docker.md)**
+**Step 1:** Docker is to slow: Make sure after installing Docker Desktop, that it has enough resources on your machine `Docker Desktop` -> `Settings` -> `Advanced`
```
Comment (sigpie) [inline webofneeds/won-docker/README.md]: As step 1 is basically "Install Docker", organising troubleshooting info by steps here may not be useful.

File: documentation/atom-structure.md [hunk 7]
```diff
@@ -67,26 +67,27 @@ atom:sitl5bx3kk96
 ## \<atom\> won:is \<description\>
```
[the remark this answers is not in the document]
Comment (m0ru) [reply documentation/atom-structure.md]: moved to `deprecated/`

File: documentation/building.md [hunk 8]
```diff
@@ -0,0 +1,422 @@
...
+
+1. Click `Import`
+2. Select the file `webofneeds/won-buildtools/src/main/resources/eclipse/formatter.xml`
+3. Click `Apply and Close`
+
+## 8. Set maven profile `skip-tests`
+
+Right-click the webofneeds project in the Project Explorer Tab on the left side >> Maven >> Select Maven Profiles. Check `skip-tests`.
+
+## 9. Maven install
```
[the remark this answers is not in the document]
Comment (m0ru) [reply documentation/building.md]: e.g. so the project-local npm exists for the next step, the required JARs exist, etc

[the remark this answers is not in the document]
Comment (sigpie) [reply documentation/building.md]: those questions are more "I'm missing this information in the documentation", less "I have no idea why this is happening"

File: documentation/configuration-logging.md (hunk 1 above)
Comment (m0ru) [reply documentation/configuration-logging.md]: let's skip that.

Comment (m0ru) [reply documentation/configuration-logging.md]: replaced with links.

**Comment by m0ru** [comment]
if the docs are alright, plz check it out and run a `mvn install` to see if the updated `pom.xml`s work for you.

**Comment by m0ru** [comment]
@fkleedorfer meant that he'd like to keep https://github.com/researchstudio-sat/webofneeds/blob/master/documentation/maintenance-monitoring-won-docker-containers.md around. As it's deprecated it should be moved to `deprecated/`.

**Pull request merged**